Updates to the Event Hub component

eavanvalkenburg opened the pull request:
## Description:
Made updates to the way HA talks to Azure Event Hub, fixing several items. It also implements config flow for AEH, including tests. Finally it adds connection string based creation of Event Hub connections, rather than the model with a namespace, instance and SAS policy and key.

**Related issue (if applicable):** fixes #25020, fixes #25695, fixes #25421

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [x] There is no commented out code in this PR.
  - [x] I have followed the [development checklist][dev-checklist]

If the code communicates with devices, web services, or third-party tools:
  - [x] [_The manifest file_][manifest-docs] has all fields filled out correctly. Update and include derived files by running `python3 -m script.hassfest`.
  - [x] New or updated dependencies have been added to `requirements_all.txt` by running `python3 -m script.gen_requirements_all`.
  - [x] Untested files have been added to `.coveragerc`.

If the code does not interact with devices:
  - [x] Tests have been added to verify that the new code works.

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html

eavanvalkenburg committed:
added new conn string setups, made filter optional (as is documented)

eavanvalkenburg committed:
aeh updates

eavanvalkenburg committed:
update to aeh

eavanvalkenburg committed:
update to aeh

eavanvalkenburg committed:
rebuilt of AEH integration

eavanvalkenburg committed:
implements config flow, thread-based aeh and tests

eavanvalkenburg committed:
forgot requirements file

eavanvalkenburg committed:
fixes to requirements again

eavanvalkenburg committed:
pylint fixes

eavanvalkenburg committed:
another round of pylint and black interaction fixes

pvizeli commented on homeassistant/components/azure_event_hub/__init__.py:
```diff
@@ -1,88 +1,260 @@
+    def _event_listener(self, event: Event):
+        """Listen for new messages on the bus and queue them for AEH."""
+        item = (time.monotonic(), event)
+        self.queue.put(item)
```
You can't write from a asyncio loop into a normal queue. I mixup the asyncio / thread a bit too much :) Decide you for one direction to work with threads or coroutine/jobs and use only async commands or sync commands. If you API only support sync calls, use threads otherwise asyncio jobs.

eavanvalkenburg replied:
thanks for having a look @pvizeli, I took most of the code around the setup for a thread from the Recorder component. I know that doesn't mean the principles used then still apply but I think now in this component, the functions surrounding setup and shutdown are async, while the rest of the running thread is sync, which kind of makes sense to me, especially in combination with a config flow (not blocking everything while setting up the Thread).

pvizeli replied:
If you don't mark this function as async with callback, it works to push the event to queue

eavanvalkenburg committed:
fix based on comment and better tests

MartinHjelmare commented:
Please split this PR into multiple separate PRs, one for the refactor, one for the config flow and one for the connection feature addition.

The smaller the PR the faster it will be merged.

eavanvalkenburg commented:
> Please split this PR into multiple separate PRs, one for the refactor, one for the config flow and one for the connection feature addition.
> 
> The smaller the PR the faster it will be merged.

@MartinHjelmare I understand the logic behind your request, but in this case it would be a lot of work to disentangle the updates since basically the whole thing is rewritten, I updated the whole loading to be triggered through the config and a bunch of code from setup was changed and moved down into the running code since the sending logic has changed, and there is then also code in the setup that allows for the use of the connection strings, so it is possible but also lots of extra work on my end, and extra time for it all to get through, since they depend on each other I then have to wait for the first one to merge into dev before being able to submit the next PR.

MartinHjelmare commented:
I understand that it might be more work for you at this point, but in total it will be less work and the changes will be merged faster. Many smaller PRs are always merged faster than a single big PR.

balloob commented:
I'm going to close this PR. If it can't be broken up, it can't be reviewed. It's 1200 lines. That's just too much.
